hpy/devel/include/cpython/hpyfunc_trampolines.h

```diff
@@ -69,4 +69,17 @@
         return _h2py(IMPL(_HPyGetContext(), _py2h(self), _py2h(obj), op)); \
     }
 
+/* With the cpython ABI, the casts from Py_buffer* to HPy_buffer*
+ * happen to Just Work, which makes things easier. */
+#define _HPyFunc_TRAMPOLINE_HPyFunc_GETBUFFERPROC(SYM, IMPL) \
+    static int SYM(PyObject *arg0, Py_buffer *arg1, int arg2) \
+    { \
+        return (IMPL(_HPyGetContext(), _py2h(arg0), (HPy_buffer*)arg1, arg2)); \
```

Comment:
Rather than casting the HPy_buffer directly to a CPython one, I would rather we kept the CPython buffer separate and used `internal` to map between the two. Otherwise there is a risk of accidentally exposing things we don't want to expose (e.g. the CPython `obj`).